stdlib/array.gr

```diff
@@ -294,7 +313,10 @@ provide let forEachi = (fn, array) => {
  * @param fn: The mapper function to call on each element, where the value returned will be used to initialize the element in the new array
  * @param array: The array to iterate
  * @returns The new array with mapped values
- *
+ * 
+ * @example
+ * Array.map(a => a * a, [> 7, 8, 9]) // [> 49, 64, 81]
```

Comment:
For this one it would probably make sense to change it to:
```suggestion
 * Array.map(a => a * a, [> 7, 8, 9]) == [> 49, 64, 81]
```

Reply:
Thanks Jake.  I specifically asked Oscar this yesterday and he said for these simple ones that an in line comment was preferred.  You have missed that discussion.  I personally do not like using the == without an assert comment and then an inline showing it is true.  I sometimes feel that the use of == alone is not good, what if it is false like when comparing strings in some languages.  The in line with the resultant Array<b> is super clear.

Reply:
I'm going to have to **strongly** disagree with you here @mahSource. As we've just seen two examples above, you used a comment to show what is printed. In this example, these values aren't printed. This is confusing! In old versions of grain, we'd print the last line of your program, so it made sense, but we switch to using these equality comparisons so it was clear the value was being returned **plus** comments are never syntax highlighted, so they really just stink.

Please use comparisons, especially equality, wherever possible.

Reply:
Thanks @phated. The ones above have an embedded print() fn in the fn....so they print separate lines in reality as they functions through the cycle.  A different case for the ones that print on one line.  Is it common to use == without an assert or showing the result of the equality?

Reply:
Yes. That has been used in all our documentation. It's the same as `1 == 1` but with whatever (arrays in this case). Assert throws when the value is wrong and returns void, so isn't really helpful for real world examples.

Reply:
Hi @phated Here is another possible issue with using an operator vs. a method with == if the reader does not understand what == is implying in the example. https://www.geeksforgeeks.org/difference-between-and-equals-method-in-java/
I understand this is one case for Java but people coming to Grain might have different perceptions of what == really is from their experience.  Somehow it would be good to make it clear imho.

Reply:
I agree with your comment on assert...could add confusion to confusion.  @phated  If there is sunk cost in the rest of Docs with the use of == already then there would be consistency problem to change that.  So it sounds like when making up examples that we should use == and that (a) the reader should assume we are suggesting that the result of the operator is true and (b) that it will work with any type (i.e. not an issue with Java e.g. with strings) and is (c) immune from value vs any by reference artifacts (may not be an issue with Grain).  Also please confirm if you want any in line type outputs in any examples.  I am just trying to understand fully before spending a lot of time to create missing examples or help clean up ones that don't make sense, e.g. using == with a void return type.

Reply:
I just saw @spotandjake super helpful summary at the top of this PR (I pasted below too)....I agree with it and will use this unless @phated has an issue with it, please let me know when you can since I have some time today to work on the next stdlib.  Thanks!
From Jake:
-- Returns a value -> uses `==`
-- Returns Void -> `comment`
-- Returns Void but modifies in place -> `multiline and assert`
Nicely done Jake!